### ISSUE TITLE
React 15.5 upgrade

### ISSUE THREAD

**Issue by Matt-Butler** [cerner/terra-core#203]
The react 15.5 release was non-passive for our packages. React is currently [locked to version 15.4.2](https://github.com/cerner/terra-core/pull/202) 

We need to update our packages to be compatible with react 15.5.

https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html

Limitations:

 - [x] oliviertassinari/babel-plugin-transform-react-remove-prop-types#87
 - [x] ReactTraining/react-router#4932
 - [x] airbnb/enzyme#875
 - [x] reactjs/react-docgen#171
 - [x] yannickcr/eslint-plugin-react#1144
 - [x] https://github.com/Hacker0x01/react-datepicker/issues/813
 - [x] yahoo/react-intl#922
 - [x] reactjs/react-docgen#174
 - [x] davidtheclark/focus-trap-react#12
 - [x] https://github.com/tajo/react-portal/pull/144
 - [x] https://github.com/yahoo/react-intl/pull/940

**Comment by Matt-Butler** [comment]
https://github.com/reactjs/react-docgen/issues/171

**Comment by Matt-Butler** [comment]
https://github.com/airbnb/enzyme/issues/875

**Comment by Matt-Butler** [comment]
I started some of the 15.5.x upgrade here: https://github.com/cerner/terra-core/tree/codemod-react-proptypes

* I updated each package to use the prop-types package.
* We still need to update to react 15.5 after our 3rd party issues are resolved.

**Comment by Matt-Butler** [comment]
https://github.com/yannickcr/eslint-plugin-react/issues/1144

**Comment by bjankord** [comment]
Looks like the [enzyme PR to add compat with react 15.5 has been merged](https://github.com/airbnb/enzyme/pull/876). Waiting on a release for that.

**Comment by bjankord** [comment]
Pull request for react-docgen is still open: https://github.com/reactjs/react-docgen/pull/172

**Comment by Matt-Butler** [comment]
The react-docgen issue has been closed

**Comment by Matt-Butler** [comment]
react-datepicker merged their fix

**Comment by ryanthemanuel** [comment]
They released a version of react-docgen (2.14.1) to fix the build issues we were seeing around their compiling changes (https://github.com/reactjs/react-docgen/issues/174).  I believe that this should allow us to push to the appropriate version of react-docgen.

**Comment by bjankord** [comment]
The [react-intl pr for adding react 15.5 compat](https://github.com/yahoo/react-intl/issues/918) has been merged. This should be our last blocker for us on adding react 15.5 compat to our components.

**Comment by Matt-Butler** [comment]
Releases to support 15.5 / 16
- [x] babel-plugin-transform-react-remove-prop-types - https://github.com/oliviertassinari/babel-plugin-transform-react-remove-prop-types/releases/tag/v0.4.1
- [ ] react-intl - https://github.com/yahoo/react-intl/pull/940 - 
- [x] react-router - https://github.com/ReactTraining/react-router/releases/tag/v3.0.4
- [x] react-docgen - https://github.com/reactjs/react-docgen/releases/tag/v2.15.0
- [x] enzyme - https://github.com/airbnb/enzyme/releases/tag/v2.8.1
- [x] react-datepicker - https://github.com/Hacker0x01/react-datepicker/releases/tag/v0.46.0
- [x] focus-trap-react - https://github.com/davidtheclark/focus-trap-react/releases/tag/3.0.3
- [x] react-portal - https://github.com/tajo/react-portal/releases/tag/v3.1.0

**Comment by Matt-Butler** [comment]
https://www.npmjs.com/package/react-addons-test-utils
^ This has also been deprecated.

**Comment by Matt-Butler** [comment]
I updated the react-datepicker dependency to 0.46.  In this version, they added a new prop called `disableDateAutoCorrection`.

**Comment by MichaelDeBoey** [comment]
`react-intl v2.3.0` is [released](https://github.com/yahoo/react-intl/pull/940) 🙂

**Comment by Matt-Butler** [comment]
After updating react-intl to 2.3.0, all of our tests now pass. 

When running the site locally, I am still seeing this warning:
```
Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.
```

I will investigate which dependency is still causing this warning.

**Comment by Matt-Butler** [comment]
After going through this exercise, I think it might be a good idea to separate out each of our components into their own webpack bundle.

- It would test the component is in isolation guaranteeing that css isn't leaking from one component to another.

- It would be easier to isolate an issue down to a single component.

**Comment by Matt-Butler** [comment]
I had missed a couple proptypes. This branch is ready to upgrade to react-15.5: https://github.com/cerner/terra-core/tree/react-15.5

There are no longer warnings and all tests are passing.

**Comment by Matt-Butler** [comment]
After terra-time-input was merged into master, my tests are failing again. This is the issue:
- [ ] https://github.com/text-mask/text-mask/issues/510

I can't merge the pull request until that issue is resolved.

**Comment by Matt-Butler** [comment]
After terra-time-input was merged into master, my tests are failing again. This is the issue:
- [ ] https://github.com/text-mask/text-mask/issues/510

I can't merge the pull request until that issue is resolved.

^ We have removed text-mask for now. Closing this issue.
